File: ChallengeBackend/src/main/java/br/com/savedra/challengebackend/repository/BannerRepository.java
package br.com.savedra.challengebackend.repository;

import br.com.savedra.challengebackend.model.Banner;
import org.springframework.data.mongodb.repository.MongoRepository;
import java.util.List;

public interface BannerRepository extends MongoRepository<Banner, String> {
    List<Banner> findBySegment(String segment);
}

File: app/src/main/java/br/com/savedra/challengecrm/data/repository/BannerRepository.kt
package br.com.savedra.challengecrm.data.repository

import br.com.savedra.challengecrm.model.Banner
import com.google.firebase.firestore.FirebaseFirestore
import kotlinx.coroutines.tasks.await

class BannerRepository {
  private val firestore = FirebaseFirestore.getInstance()

  suspend fun sendBanner(banner: Banner) {
    firestore.collection("banners").add(banner).await()
  }

  suspend fun getBanners(userSegment: String? = null): List<Banner> {
    val query = if (userSegment != null) {
        firestore.collection("banners").whereIn("segment", listOf(userSegment, "Todos"))
    } else {
        firestore.collection("banners")
    }
    val result = query.get().await()
    return result.map { document ->
        document.toObject(Banner::class.java)
    }
  }
}
